module-system/sov-modules-api/src/lib.rs

```diff
@@ -340,6 +340,24 @@ pub trait Module {
     }
 }
 
+impl Event for () {
+    fn event_key(&self) -> &'static str {
+        tracing::warn!("Trying to get the event_key string for an undefined event.");
+        "NA-Event-Not-Defined"
+    }
+}
+
+/// Every module `Event` has to implement this trait.
+///
+/// You can use `#[derive(Event)]` to automatically implement the trait. If your `Event` is not an enum,
+/// or if your `Event` type is a nested enum and you want to index events based on their nested fields,
+/// you will need to implement the trait manually.
+pub trait Event: Debug + BorshSerialize + BorshDeserialize {
+    /// Returns a static string slice that identifies the Event variant.
+    /// This string slice is primarily used to index Events from the DB.
+    fn event_key(&self) -> &'static str;
```

Comment:
Is the `event_key(&self)` going to be used also for quering the database (to get all events related to a particlualar keys)?  If yes then a user will need to generate an `Event` value which is not straight forward is the venet struct/enum is complicated.

Reply:
Yeah, great point. 

Shall we have some cli tooling to generate this? I haven't looked into sov-cli much but I can imagine that being able to printing event_key's for event variants.

@preston-evans98